operator/cmd/mesh/profile-dump.go

```diff
@@ -68,17 +91,55 @@ func profileDump(args []string, rootArgs *rootArgs, pdArgs *profileDumpArgs, l *
 		}
 	}
 
-	y, _, err := GenerateConfig(pdArgs.inFilenames, setFlagYAML, true, nil, l)
-	if err != nil {
-		return err
+	includeDefaults := true
+	if pdArgs.hideInherited {
+		// only hide inherited values if the user requested it and isn't using the --config-path option.
+		includeDefaults = false
 	}
 
-	y, err = tpath.GetConfigSubtree(y, pdArgs.configPath)
+	y, _, err := GenerateConfig(pdArgs.inFilenames, setFlagYAML, true, includeDefaults, nil, l)
 	if err != nil {
 		return err
 	}
 
-	l.print(y + "\n")
+	if pdArgs.configPath == "" {
+		metaHeader := `apiVersion: install.istio.io/v1alpha2
+kind: IstioControlPlane
+spec:`
+		// indent spec by two space and prepend the k8s metadata header
+		indentedSpec := strings.ReplaceAll("\n"+y, "\n", "\n  ")
```

Comment:
suggest using tpath.AddSpecRoot since performance is not critical. also would be good to farm this and the jsonoutput logic out to helper function to limit the length of the function and help readability.

Reply:
ack